pandas/core/arrays/numpy_.py

```diff
@@ -0,0 +1,461 @@
+from functools import wraps
+import numbers
+
+import numpy as np
+
+from pandas._libs import lib
+
+from pandas.core.dtypes.common import extract_array
+from pandas.core.dtypes.dtypes import ExtensionDtype
+from pandas.core.dtypes.generic import ABCIndexClass, ABCSeries
+from pandas.core.dtypes.inference import is_list_like
+
+from pandas import compat
+from pandas.core import nanops
+
+from .base import ExtensionArray, ExtensionOpsMixin
+
+
+def _validate_reduction_args(func):
```

Comment:
@gfyoung I'd like to replace this with our existing `nv.validate_reduction_args`.

Right now if I change `all` to use it with

```python
    def all(self, axis=None, out=None, keepdims=None, skipna=True):
        nv.validate_all((), dict(out=out, keepdims=keepdims))
        return nanops.nanall(self._ndarray, axis=axis, skipna=skipna)
```

I get a  `TypeError: all() got an unexpected keyword argument 'keepdims'`

Do you know, should our `validate_*` functions handle `keepdims`?

Other keywords I'd like to validate:

- `initial` (sum)
- `overwrite_input` (median)
- `dtype` (several methods)